jans-config-api/plugins/scim-plugin/src/main/java/io/jans/configapi/plugin/scim/configuration/ScimConfigurationFactory.java

```diff
@@ -0,0 +1,212 @@
+import io.jans.orm.model.PersistenceConfiguration;
+import io.jans.orm.service.PersistanceFactoryService;
+import io.jans.service.cdi.event.ConfigurationUpdate;
+import io.jans.util.properties.FileConfiguration;
+
+import java.io.File;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import javax.annotation.PostConstruct;
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.event.Event;
+import javax.enterprise.inject.Produces;
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+
+@ApplicationScoped
+public class ScimConfigurationFactory {
+    
+    public static final String CONFIGURATION_ENTRY_DN = "scim_ConfigurationEntryDN";
+
+    @Inject
+    private Logger log;
+
+    @Inject
+    private Event<ScimAppConfiguration> configurationUpdateEvent;
+
+    @Inject
+    private PersistanceFactoryService persistanceFactoryService;
+
+    @Inject
+    private PersistenceEntryManager entryManager;
+
+    public final static String PERSISTENCE_CONFIGUARION_RELOAD_EVENT_TYPE = "persistenceConfigurationReloadEvent";
+    public final static String BASE_CONFIGUARION_RELOAD_EVENT_TYPE = "baseConfigurationReloadEvent";
+
+    static {
+        if (System.getProperty("jans.base") != null) {
+            BASE_DIR = System.getProperty("jans.base");
+        } else if ((System.getProperty("catalina.base") != null) && (System.getProperty("catalina.base.ignore") == null)) {
+            BASE_DIR = System.getProperty("catalina.base");
+        } else if (System.getProperty("catalina.home") != null) {
+            BASE_DIR = System.getProperty("catalina.home");
+        } else if (System.getProperty("jboss.home.dir") != null) {
+            BASE_DIR = System.getProperty("jboss.home.dir");
+        } else {
+            BASE_DIR = null;
+        }
+    }
+
+    private static final String BASE_DIR;
+    private static final String DIR = BASE_DIR + File.separator + "conf" + File.separator;
+    private static final String BASE_PROPERTIES_FILE = DIR + "jans.properties";
+    private static final String APP_PROPERTIES_FILE = DIR + "scim.properties";
+
+
+    private boolean loaded = false;
+    private FileConfiguration baseConfiguration;    
+    private PersistenceConfiguration persistenceConfiguration;
+    private ScimAppConfiguration dynamicConf;
+    private StaticConfiguration staticConf;
+    private String cryptoConfigurationSalt;
+
+    private AtomicBoolean isActive;
+    private long baseConfigurationFileLastModifiedTime;
+    private long loadedRevision = -1;
+    private boolean loadedFromLdap = true;
+
+    @PostConstruct
+    public void init() {
+        this.isActive = new AtomicBoolean(true);
+        try {
+            log.error("ScimConfigurationFactory::loadConfigurationFromLdap() - persistanceFactoryService:{} ",persistanceFactoryService);
+            this.persistenceConfiguration = persistanceFactoryService.loadPersistenceConfiguration(APP_PROPERTIES_FILE);
+            
+            loadBaseConfiguration();
```

Comment:
It seems we load configuration from file and don't load it from db.
Don't we need to load it from db ? It sounds as `create()` method should not be commented.

Reply:
We load it from underlying persistence via [Conf](https://github.com/JanssenProject/jans/blob/jans-config-api-issue-271/jans-config-api/plugins/scim-plugin/src/main/java/io/jans/configapi/plugin/scim/service/ScimConfigService.java)